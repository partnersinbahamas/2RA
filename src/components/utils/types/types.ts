export type TSize = 'small' | 'medium' | 'large';

export type TStile = 'primary' | 'default' | 'mute';

export type TStatus = "online" | "default" | "busy" | "not-here" | "sleep" | "primary";
export type TStringOrStatus = TStatus | string;
