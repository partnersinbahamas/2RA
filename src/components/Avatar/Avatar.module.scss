@import '../utils/colors.scss';


// component structur
[data-name="badge"] {
  color: $white;
  min-width: 0.625rem;
  min-height: 0.625rem;
  border: 1px solid transparent;
  border-radius: 50%;

  display: flex;
  align-items: center;
  justify-content: center;
}

[data-status="online"] {
  background-color: $green_400;
}

[data-status="default"] {
  background-color: $cool_gray_200;
  border-color: $cool_gray_400;
}

[data-status="sleep"] {
  background-color: $violet_600;
}

[data-status="busy"] {
  background-color: $red_400;
}

[data-status="not-here"] {
  background-color: $orange_400;
}

[data-name="avatar"] {
  display: flex;
  align-items: center;
  justify-content: center;
  border-radius: 50%;
  background-color: $cool_gray_200;
}

//classes
.primary {
  [data-name="avatar"] {
    background-color: $cyan_400;
    border: 3px solid $blue_400;
    font-weight: 600;
  }

  [data-status="primary"] {
    background-color: $blue_400;
  }
};

div.small {
  width: 1.25rem;
  height: 1.25rem;
  font-size: 0.5rem;
}

div.medium {
  width: 2.188rem;
  height: 2.188rem;
  font-size: 0.875rem;
}

div.large {
  width: 3.125rem;
  height: 3.125rem;
  font-size: 1.25rem;
}