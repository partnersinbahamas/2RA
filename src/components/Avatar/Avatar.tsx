import React from 'react';
import { Avatar as AvatarMUI, AvatarProps, AvatarSlots, Badge, Box } from '@mui/material';
import classNames from 'classnames';

import { PropsAvatar } from '../utils/types/props/avatar';
import { TStringOrStatus } from '../utils/types/types';
import defaultProps from '../utils/variables/defaultProps';

import styles from './Avatar.module.scss';

interface IProps extends PropsAvatar {
  /**
   * Instead of setting first and last name, you can also set an abbreviation prop.
   */
  abbreviation: string;
  /**
   * Path to image.
   */
  src?: string;
  /**
   * Details on MaterialUI.
   */
  slots?: AvatarSlots;
  /**
   * You definitely can attach your own statuses and style them.
   */
  status?: TStringOrStatus;
}

type TAvatarComponent = AvatarProps & TExtendedComponentProps
const AvatarComponent: React.FC<TAvatarComponent> = (props) => <AvatarMUI {...props} />

export const Avatar: React.FC<IProps> = ({
  firstName,
  lastName,
  abbreviation,
  backgroundColor,
  badged,
  disabled,
  badgeContent,
  anchorOrigin = defaultProps.anchorOrigin,
  size = defaultProps.size,
  status = defaultProps.status,
  stile = defaultProps.stile,
  onClick,
  src,
  ...props
}) => {
  const { className } = props;
  const fullName = `${firstName} ${lastName}`;

  if (!badged) {
    return (
      <div className={`${className} ${styles[stile]}`}>
        <AvatarComponent
          {...props}
          src={src}
          data-name="avatar"
          data-testid="avatar"
          data-stile={stile && stile}
          aria-disabled={disabled}
          alt={fullName}
          sx={{ backgroundColor }}
          onClick={onClick}
          className={classNames(styles[size], `${className}-avatar`)}
        >
          <span
            data-name="abbreviation"
            data-testid="avatar-abbreviation"
            className={`${className}-abbreviation`}
          >
            {firstName && lastName ? `${firstName[0]}${lastName[0]}` : abbreviation}
          </span>
        </AvatarComponent>
      </div>
    );
  }

  return (
    <Badge
      className={`${className} ${styles[stile]}`}
      overlap="circular"
      badgeContent={
        <Box
          data-testid="badge"
          data-name="badge"
          data-status={status}
          aria-disabled={disabled}
          className={`${className}-badge ${className}-badge-${status}`}
        >
          {badgeContent}
        </Box>
      }
      anchorOrigin={anchorOrigin}
    >
      <AvatarComponent
        {...props}
        src={src}
        data-name="avatar"
        data-testid="avatar"
        aria-disabled={disabled}
        data-stile={stile && stile}
        alt={fullName}
        sx={{ backgroundColor }}
        onClick={onClick}
        className={classNames(styles[size], `${className}-avatar`)}
      >
        <span
          data-name="abbreviation"
          data-testid="avatar-abbreviation"
          className={`${className}-abbreviation`}
        >
          {firstName && lastName ? `${firstName[0]}${lastName[0]}` : abbreviation}
        </span>
      </AvatarComponent>
    </Badge>
  );
};
