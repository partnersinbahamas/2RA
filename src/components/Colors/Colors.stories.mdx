import { Meta, ColorPalette, ColorItem } from '@storybook/blocks';
import * as col from './colors.ts';

<Meta title="Design Tokens/Colors" />

<ColorPalette>
  <ColorItem
    title="Black"
    colors={{
      BLACK: col.BLACK,
    }}
  />
  <ColorItem
    title="White"
    colors={{
      WHITE: col.WHITE,
    }}
  />
  <ColorItem
    title="Primary"
    colors={{
      RED_500: col.RED_500,
    }}
  />
  <ColorItem
    title="Secondary"
    colors={{
      COOL_GRAY_400: col.COOL_GRAY_400,
    }}
  />
  <ColorItem
    title="Yellow"
    colors={{
      YELLOW_100: col.YELLOW_100,
      YELLOW_200: col.YELLOW_200,
      YELLOW_300: col.YELLOW_300,
      YELLOW_400: col.YELLOW_400,
      YELLOW_500: col.YELLOW_500,
      YELLOW_600: col.YELLOW_600,
      YELLOW_700: col.YELLOW_700,
      YELLOW_800: col.YELLOW_800,
    }}
  />
  <ColorItem
    title="Orange"
    colors={{
      ORANGE_100: col.ORANGE_100,
      ORANGE_200: col.ORANGE_200,
      ORANGE_300: col.ORANGE_300,
      ORANGE_400: col.ORANGE_400,
      ORANGE_500: col.ORANGE_500,
      ORANGE_600: col.ORANGE_600,
      ORANGE_700: col.ORANGE_700,
      ORANGE_800: col.ORANGE_800,
    }}
  />
  <ColorItem
    title="Red"
    colors={{
      RED_100: col.RED_100,
      RED_200: col.RED_200,
      RED_300: col.RED_300,
      RED_400: col.RED_400,
      RED_500: col.RED_500,
      RED_600: col.RED_600,
      RED_700: col.RED_700,
      RED_800: col.RED_800,
    }}
  />
  <ColorItem
    title="Burgundy"
    colors={{
      BURGUNDY_100: col.BURGUNDY_100,
      BURGUNDY_200: col.BURGUNDY_200,
      BURGUNDY_300: col.BURGUNDY_300,
      BURGUNDY_400: col.BURGUNDY_400,
      BURGUNDY_500: col.BURGUNDY_500,
      BURGUNDY_600: col.BURGUNDY_600,
      BURGUNDY_700: col.BURGUNDY_700,
      BURGUNDY_800: col.BURGUNDY_800,
    }}
  />
  <ColorItem
    title="Pink"
    colors={{
      PINK_100: col.PINK_100,
      PINK_200: col.PINK_200,
      PINK_300: col.PINK_300,
      PINK_400: col.PINK_400,
      PINK_500: col.PINK_500,
      PINK_600: col.PINK_600,
      PINK_700: col.PINK_700,
      PINK_800: col.PINK_800,
    }}
  />
  <ColorItem
    title="Violet"
    colors={{
      VIOLET_100: col.VIOLET_100,
      VIOLET_200: col.VIOLET_200,
      VIOLET_300: col.VIOLET_300,
      VIOLET_400: col.VIOLET_400,
      VIOLET_500: col.VIOLET_500,
      VIOLET_600: col.VIOLET_600,
      VIOLET_700: col.VIOLET_700,
      VIOLET_800: col.VIOLET_800,
    }}
  />
  <ColorItem
    title="Blue"
    colors={{
      BLUE_100: col.BLUE_100,
      BLUE_200: col.BLUE_200,
      BLUE_300: col.BLUE_300,
      BLUE_400: col.BLUE_400,
      BLUE_500: col.BLUE_500,
      BLUE_600: col.BLUE_600,
      BLUE_700: col.BLUE_700,
      BLUE_800: col.BLUE_800,
    }}
  />
  <ColorItem
    title="Cyan"
    colors={{
      CYAN_100: col.CYAN_100,
      CYAN_200: col.CYAN_200,
      CYAN_300: col.CYAN_300,
      CYAN_400: col.CYAN_400,
      CYAN_500: col.CYAN_500,
      CYAN_600: col.CYAN_600,
      CYAN_700: col.CYAN_700,
      CYAN_800: col.CYAN_800,
    }}
  />
  <ColorItem
    title="Turquoise"
    colors={{
      TURQUOISE_100: col.TURQUOISE_100,
      TURQUOISE_200: col.TURQUOISE_200,
      TURQUOISE_300: col.TURQUOISE_300,
      TURQUOISE_400: col.TURQUOISE_400,
      TURQUOISE_500: col.TURQUOISE_500,
      TURQUOISE_600: col.TURQUOISE_600,
      TURQUOISE_700: col.TURQUOISE_700,
      TURQUOISE_800: col.TURQUOISE_800,
    }}
  />
  <ColorItem
    title="Green"
    colors={{
      GREEN_100: col.GREEN_100,
      GREEN_200: col.GREEN_200,
      GREEN_300: col.GREEN_300,
      GREEN_400: col.GREEN_400,
      GREEN_500: col.GREEN_500,
      GREEN_600: col.GREEN_600,
      GREEN_700: col.GREEN_700,
      GREEN_800: col.GREEN_800,
    }}
  />
  <ColorItem
    title="Light Green"
    colors={{
      LIGHT_GREEN_100: col.LIGHT_GREEN_100,
      LIGHT_GREEN_200: col.LIGHT_GREEN_200,
      LIGHT_GREEN_300: col.LIGHT_GREEN_300,
      LIGHT_GREEN_400: col.LIGHT_GREEN_400,
      LIGHT_GREEN_500: col.LIGHT_GREEN_500,
      LIGHT_GREEN_600: col.LIGHT_GREEN_600,
      LIGHT_GREEN_700: col.LIGHT_GREEN_700,
      LIGHT_GREEN_800: col.LIGHT_GREEN_800,
    }}
  />
  <ColorItem
    title="Warm Gray"
    colors={{
      WARM_GRAY_100: col.WARM_GRAY_100,
      WARM_GRAY_200: col.WARM_GRAY_200,
      WARM_GRAY_300: col.WARM_GRAY_300,
      WARM_GRAY_400: col.WARM_GRAY_400,
      WARM_GRAY_500: col.WARM_GRAY_500,
      WARM_GRAY_600: col.WARM_GRAY_600,
      WARM_GRAY_700: col.WARM_GRAY_700,
      WARM_GRAY_800: col.WARM_GRAY_800,
    }}
  />
  <ColorItem
    title="Cool Gray"
    colors={{
      COOL_GRAY_100: col.COOL_GRAY_100,
      COOL_GRAY_200: col.COOL_GRAY_200,
      COOL_GRAY_300: col.COOL_GRAY_300,
      COOL_GRAY_400: col.COOL_GRAY_400,
      COOL_GRAY_500: col.COOL_GRAY_500,
      COOL_GRAY_600: col.COOL_GRAY_600,
      COOL_GRAY_700: col.COOL_GRAY_700,
      COOL_GRAY_800: col.COOL_GRAY_800,
    }}
  />
</ColorPalette>
