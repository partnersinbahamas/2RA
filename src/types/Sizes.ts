export type TSizes = 'small' | 'medium' | 'large';