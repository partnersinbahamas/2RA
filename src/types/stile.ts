export type TStile = 'primary' | 'default' | 'mute';