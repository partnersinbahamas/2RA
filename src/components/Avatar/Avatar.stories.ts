import { Meta, StoryObj } from '@storybook/react';

import { Avatar } from '../Avatar/Avatar';
import me from '../../images/me.png';

/**
 * Avatar conponent with user picture and fullname.
 */

const meta = {
  title: 'Athomic/Avatar',
  component: Avatar,
  tags: ['autodocs'],
  parameters: {
    layout: 'centered',
  },
  args: {
    abbreviation: 'EM',
  },
  argTypes: {
    firstName: { control: 'text' },
    lastName: { control: 'text' },
    abbreviation: { control: 'text' },
    backgroundColor: { control: 'color' },
    badged: { control: 'boolean' },
    badgeContent: { control: 'text' },
    src: { control: 'text' },
    onClick: { control: 'changed' },
    stile: { control: 'inline-radio', options: ['primary'] },
    size: {
      control: 'inline-radio',
      options: ['small', 'medium', 'large'],
    },
    status: {
      control: 'inline-radio',
      options: ['online', 'default', 'not-here', 'busy', 'sleep', 'primary'],
    },
  }
} satisfies Meta<typeof Avatar>;
export default meta;
type TStory = StoryObj<typeof meta>;

export const Default: TStory = {
  args: {
    size: 'medium',
    firstName: 'Steve',
    lastName: 'Jobs',
  }
};
export const Primary: TStory = {
  args: {
    size: 'medium',
    stile: 'primary',
    status: 'primary',
    badged: true,
  }
}
export const Badged: TStory = {
  args: {
    size: 'medium',
    badged: true,
  }
};
export const Online: TStory = {
  args: {
    size: 'medium',
    status: 'online',
    badged: true,
  }
};
// status stories
export const NotHere: TStory = {
  args: {
    size: 'medium',
    status: 'not-here',
    badged: true,
  }
};
export const Busy: TStory = {
  args: {
    size: 'medium',
    status: 'busy',
    badged: true,
  }
};
export const Sleep: TStory = {
  args: {
    size: 'medium',
    status: 'sleep',
    badged: true,
  }
};
// status stories
export const BackgroundImage: TStory = {
  args: {
    size: 'large',
    src: me,
  }
};