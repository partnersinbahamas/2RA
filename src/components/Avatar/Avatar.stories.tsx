import { Meta, StoryObj } from '@storybook/react';

import HDLogo from '../../images/logos/hd-logo-dark-small.png';
import { Avatar } from './Avatar';
import { story } from '../utils/variables/story';
import { ArgTypes } from 'storybook/internal/types';
import { DocsPage } from '@storybook/addon-docs';
import { Markdown, Source, Subtitle } from '@storybook/blocks';

/**
 * Avatar conponent with user picture and fullname.
 */

const AVATAR_SCSS_STYLES = `
.component {
  .component-avatar {
    &[disabled] {}
    &[error] {};
  };

  .component-badge {
    &[status="statusProp"] {
      &[disabled] {};
      &[error] {};
    };
  };

  .component-abbreviation {
    &[error] {};
    &[disabled] {};
  };
}`;

const AVATAR_MODULE_SCSS_STYLES = `
div.avatar {
  &[disabled] {};
  &[error] {};
};

.badge {
  &[status="statusProp"] {
    &[disabled] {};
    &[error] {};
  };
};

.abbreviation {
  &[error] {};
  &[disabled] {};
}`;

const meta = {
  title: 'Athomic/Avatar',
  component: Avatar,
  tags: ['autodocs'],
  args: {
    abbreviation: 'HD',
    anchorOrigin: { horizontal: 'right', vertical: 'bottom' },
    stile: 'default',
  },
  parameters: {
    layout: 'centered',
    docs: {
      page: () => {
        return (
          <>
            <Subtitle>
              Please to style your custom Avatar component use the mockup below.
            </Subtitle>
            <Markdown>Guide styles | .module.scss</Markdown>
            <Source language="css" code={AVATAR_MODULE_SCSS_STYLES} />

            <Markdown>Guide styles | .scss</Markdown>
            <Source language="css" code={AVATAR_SCSS_STYLES} />
            <DocsPage />
          </>
        );
      },
    },
  },
  argTypes: {
    firstName: story.name,
    lastName: story.name,
    abbreviation: story.abbreviation,
    badged: story.badged,
    disabled: story.disabled,
    src: story.src,
    onClick: story.onClick,
    stile: story.stile,
    size: story.size,
    status: story.status,
    badgeContent: { control: 'text' },
    error: story.stringError,
    anchorOrigin: {
      control: {
        type: 'object', // indicate that is an object
        keys: {
          horizontal: {
            type: 'select',
            options: ['left', 'right'],
          },
          vertical: {
            type: 'select',
            options: ['top', 'bottom'],
          },
        },
      },
    },
  } as ArgTypes,
} satisfies Meta<typeof Avatar>;
export default meta;
type TStory = StoryObj<typeof meta>;

export const Default: TStory = {
  args: {
    size: 'medium',
  },
};
export const Muted: TStory = {
  args: {
    size: 'medium',
    stile: 'mute',
  },
};
export const Disabled: TStory = {
  args: {
    size: 'medium',
    disabled: true,
    badged: true,
  },
};
export const Error: TStory = {
  args: {
    size: 'medium',
    badged: true,
    error: true,
  },
};
export const Primary: TStory = {
  args: {
    size: 'medium',
    stile: 'primary',
    status: 'primary',
    badged: true,
  },
};
export const PrimaryDisabled: TStory = {
  args: {
    size: 'medium',
    stile: 'primary',
    status: 'primary',
    badged: true,
    disabled: true,
  },
};
export const PrimaryError: TStory = {
  args: {
    size: 'medium',
    stile: 'primary',
    status: 'primary',
    badged: true,
    error: true,
  },
};
export const Badged: TStory = {
  args: {
    size: 'medium',
    badged: true,
  },
};
export const Online: TStory = {
  args: {
    size: 'medium',
    status: 'online',
    badged: true,
  },
};
// status stories
export const NotHere: TStory = {
  args: {
    size: 'medium',
    status: 'not-here',
    badged: true,
  },
};
export const Busy: TStory = {
  args: {
    size: 'medium',
    status: 'busy',
    badged: true,
  },
};
export const Sleep: TStory = {
  args: {
    size: 'medium',
    status: 'sleep',
    badged: true,
  },
};
// status stories
export const BackgroundImage: TStory = {
  args: {
    size: 'large',
    src: HDLogo,
  },
};
