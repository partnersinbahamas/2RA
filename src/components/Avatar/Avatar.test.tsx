import userEvent from '@testing-library/user-event';
import { render, screen } from '@testing-library/react';

import { YELLOW_600 } from '../Colors/colors';
import { TSize } from '../utils/types/types';
import { Avatar } from './Avatar';

describe('Avatar', () => {
  const user = userEvent.setup();

  const abbreviation = 'SJ';
  const size: TSize = 'medium';

  it('renders correclty', () => {
    const { container } = render(<Avatar size={size} abbreviation={abbreviation} />);
    const avatarElement = screen.getByTestId('avatar');
    expect(avatarElement).toBeInTheDocument();
    expect(avatarElement).toHaveClass('medium');

    const badgeElement = screen.queryByTestId('badge');
    expect(badgeElement).toBeNull();

    const nameElement = screen.getByTestId('avatar-abbreviation');
    expect(nameElement).toHaveTextContent(abbreviation);
    expect(container).toMatchSnapshot();
  });
  it('renders with background color', () => {
    const bgc = YELLOW_600;
    const { container } = render(
      <Avatar size={size} abbreviation={abbreviation} backgroundColor={bgc} />,
    );
    const avatarElement = screen.getByTestId('avatar');
    expect(avatarElement).toHaveStyle(`backgroundColor: ${bgc}`);
    expect(container).toMatchSnapshot();
  });
  it('renders on click', async () => {
    const onClickHandler = jest.fn();

    render(<Avatar size={size} abbreviation={abbreviation} onClick={onClickHandler} />);

    const avatarElement = screen.getByTestId('avatar');
    await user.click(avatarElement);
    expect(onClickHandler).toHaveBeenCalled();
  });
  it('renders with stile primary', () => {
    const { container } = render(
      <Avatar size={size} abbreviation={abbreviation} stile="primary" />,
    );

    const avatarElement = screen.getByTestId('avatar');
    expect(avatarElement.getAttribute('data-stile')).toBe('primary');
    expect(container).toMatchSnapshot();
  });
  it('renders with badged', () => {
    const content = 'Badge content';
    const { container } = render(
      <Avatar size={size} abbreviation={abbreviation} badged badgeContent={content} />,
    );

    const badgeElement = screen.queryByTestId('badge');
    expect(badgeElement).not.toBeNull();

    expect(badgeElement).toHaveTextContent(content);

    expect(container).toMatchSnapshot();
  });

  it('renders with name', () => {
    const firstName = 'Elon';
    const lastName = 'Musk';
    const fullNameAbbreviation = `${firstName[0]}${lastName[0]}`;
    const { container } = render(
      <Avatar size={size} abbreviation={abbreviation} firstName={firstName} lastName={lastName} />,
    );
    const avatarNameElement = screen.getByTestId('avatar-abbreviation');
    expect(avatarNameElement).toHaveTextContent(fullNameAbbreviation);

    expect(container).toMatchSnapshot();
  });

  it('renders on disabled', () => {
    const { container } = render(
      <Avatar size={size} abbreviation={abbreviation} disabled badged />,
    );

    const badgeElement = screen.getByTestId('badge');
    expect(badgeElement.getAttribute('aria-disabled')).toEqual('true');
    const avatarElement = screen.getByTestId('avatar');
    expect(avatarElement.getAttribute('aria-disabled')).toEqual('true');

    expect(container).toMatchSnapshot();
  });

  it('renders with className props', () => {
    const className = 'avatar-className';
    const { container } = render(
      <Avatar size={size} abbreviation={abbreviation} badged className={className} />,
    );
    const avatarElement = screen.getByTestId('badge');
    expect(avatarElement).toHaveClass(`${className}-badge`);
    expect(container).toMatchSnapshot();
  });
});
