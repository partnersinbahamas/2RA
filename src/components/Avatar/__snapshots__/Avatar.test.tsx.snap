// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar renders correclty 1`] = `
<div>
  <div
    class="undefined undefined"
  >
    <div
      class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault medium css-17o22dy-MuiAvatar-root"
      data-name="avatar"
      data-testid="avatar"
    >
      <span
        data-name="name"
        data-testid="avatar-name"
      >
        SJ
      </span>
    </div>
  </div>
</div>
`;

exports[`Avatar renders with background color 1`] = `
<div>
  <div
    class="undefined undefined"
  >
    <div
      class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault medium css-ww4vf0-MuiAvatar-root"
      data-name="avatar"
      data-testid="avatar"
    >
      <span
        data-name="name"
        data-testid="avatar-name"
      >
        SJ
      </span>
    </div>
  </div>
</div>
`;

exports[`Avatar renders with badged 1`] = `
<div>
  <span
    class="MuiBadge-root undefined undefined css-1c32n2y-MuiBadge-root"
  >
    <div
      class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault medium css-17o22dy-MuiAvatar-root"
      data-name="avatar"
      data-testid="avatar"
    >
      <span
        data-name="name"
        data-testid="avatar-name"
      >
        SJ
      </span>
    </div>
    <span
      class="MuiBadge-badge MuiBadge-standard MuiBadge-anchorOriginBottomRight MuiBadge-anchorOriginBottomRightCircular MuiBadge-overlapCircular css-9sufak-MuiBadge-badge"
    >
      <div
        class="MuiBox-root css-0"
        data-name="badge"
        data-status="default"
        data-testid="badge"
      >
        Badge content
      </div>
    </span>
  </span>
</div>
`;

exports[`Avatar renders with name 1`] = `
<div>
  <div
    class="undefined undefined"
  >
    <div
      class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault medium css-17o22dy-MuiAvatar-root"
      data-name="avatar"
      data-testid="avatar"
    >
      <span
        data-name="name"
        data-testid="avatar-name"
      >
        EM
      </span>
    </div>
  </div>
</div>
`;

exports[`Avatar renders with stile primary 1`] = `
<div>
  <div
    class="undefined primary"
  >
    <div
      class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault medium css-17o22dy-MuiAvatar-root"
      data-name="avatar"
      data-stile="primary"
      data-testid="avatar"
    >
      <span
        data-name="name"
        data-testid="avatar-name"
      >
        SJ
      </span>
    </div>
  </div>
</div>
`;
